### DISCUSSION TITLE
How to avoid typescript inside the bundle

### DISCUSSION THREAD

**Discussion by manoraj** [oclif/core#1236] [General]
Hi, 

We are using a [bundling strategy](https://oclif.io/docs/command_discovery_strategies#bundling) in our CLI. With all the commands, the size of the bundle is around 4.3MB. We have noticed that the bundle includes a typescript which is 3.3MB. Excluding this from the bundle will give us great startup improvements. 

Question:

1. Does @oclif/core need a typescript package for bundling strategy? If yes, why?
2. Can we disable it?

![Screenshot 2024-10-29 at 11 24 54 PM](https://github.com/user-attachments/assets/b2233fcd-ea03-4c5a-bb9f-4ce4065288d2)
![Screenshot 2024-10-29 at 11 24 59 PM](https://github.com/user-attachments/assets/8d1f037a-6e09-4d20-a87f-90edb91a0b53)

**Comment by mdonnalley** [comment]
@manoraj @oclif/core doesn't list typescript as a dependency in any way. In fact, we have a [test](https://github.com/oclif/core/actions/runs/11562497390/job/32183816329) that ensures it's never included

My guess is that it's being included in your bundle because of [this line](https://github.com/oclif/core/blob/4.0.31/src/util/read-tsconfig.ts#L40). That code is meant to support auto-compiling a plugin in an development environment - the assumption is that typescript will be a devDependency of the plugin you're developing. If it can't find typescript, it'll bypass the auto-compilation step and use whatever compiled source is available to run.

Based on that, I would configure your bundler to skip or remove typescript from the bundle (I'm assuming that's possible - I don't have a lot of experience bundling). Your CLI will be able to operate just fine without typescript in the bundle

**Reply by manoraj** [reply]
Hi @mdonnalley Thanks for the clarification. Let me check this and get back with my findings
